src/message.c

```diff
@@ -1060,14 +1070,95 @@ delete_first_msg(void)
     return OK;
 }
 
-    void
+    static void
 check_msg_hist(void)
 {
     // Don't let the message history get too big
-    while (msg_hist_len > 0 && msg_hist_len > p_mhi)
+    while (msg_hist_len > 0 && msg_hist_len > msg_hist_max)
 	(void)delete_first_msg();
 }
 
+
+    int
+messagesopt_changed(void)
+{
+    char_u	*p;
+    int		messages_flags_new = 0;
+    int		messages_wait_new = 0;
+    int		messages_history_new = 0;
+
+    const char HIT_ENTER[] = "hit-enter";
+    const char WAIT[] = "wait:";
+    const char HISTORY[] = "history:";
+
+    const int LEN_HIT_ENTER = STRLEN_LITERAL(HIT_ENTER);
+    const int LEN_WAIT = STRLEN_LITERAL(WAIT);
+    const int LEN_HISTORY = STRLEN_LITERAL(HISTORY);
```

Comment:
You are confusing the language specifications of C and C++.
In the case of C language, even if you add "const", it doesn't become a constant.
Every time the function is called, an area is assigned on the stack and the value is set.
You can confirm this by adding the "-S" option at compile time to output the assembler code.

Why don't you simply follow `diffopt_changed()`?
If you have reservations about using immediate values, use a "#define" macro.